### PR TITLE
Nightmare's Lighteater now smashes the lamp of the plamsaman helmets instead of dusting them.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes the Nightmare's Lighteater destroy the lamp of the plasmaman helmet if it's on, instead of dusting the helmet outright.

## Why It's Good For The Game
Kinda sucks to get jumpscared by a Nightmare and loose your only helmet while playing as a plasmaman, which odds are will end up with you husked.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/110184118/216715321-c7e39683-9a61-430b-b582-581d22b80952.mp4

</details>

## Changelog
:cl:
tweak: Instead of being dusted, plasmaman helmets now have their lamps broken when hit with the Light Eater
add: Added the ability to repair broken plasmaman helmet lamps with light bulbs
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
This isn't a bug, it's a hilarious feature.

**Comment by CydiaLamiales** [comment]
And I didn't imply it's a bug, it's more of an oversight.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 1]
```diff
@@ -210,6 +219,20 @@
 	user.update_inv_head() //So the mob overlay updates
 	update_button_icons(user)
 
+/obj/item/clothing/head/helmet/space/plasmaman/proc/smash_headlamp()
+	if(!lamp_functional)
+		return
+	if(!helmet_on)
```
Comment (itsmeow) [inline code/modules/clothing/spacesuits/plasmamen.dm:225]: can the helmet emit light if it's on the ground?

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 2]
```diff
@@ -107,6 +108,10 @@
 
 /obj/item/clothing/head/helmet/space/plasmaman/attackby(obj/item/item, mob/living/user)
 	. = ..()
+	if(istype(item, /obj/item/flashlight) && !lamp_functional)
```
Comment (itsmeow) [inline code/modules/clothing/spacesuits/plasmamen.dm]: maybe just a light bulb instead

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 3]
```diff
@@ -195,7 +200,11 @@
 
 /obj/item/clothing/head/helmet/space/plasmaman/attack_self(mob/user)
 	helmet_on = !helmet_on
-
+	if(!lamp_functional)
+		to_chat(user, "<span class='notice'>Your helmet's torch is broken!</span>")
```
Comment (itsmeow) [inline code/modules/clothing/spacesuits/plasmamen.dm]: Mention putting a new bulb or flashlight maybe

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 1 above)
Comment (CydiaLamiales) [reply code/modules/clothing/spacesuits/plasmamen.dm:225]: Yes

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 2 above)
Comment (CydiaLamiales) [inline code/modules/clothing/spacesuits/plasmamen.dm]:
```suggestion
	if(istype(item, /obj/item/light/bulb) && !lamp_functional)
```

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 3 above)
Comment (CydiaLamiales) [inline code/modules/clothing/spacesuits/plasmamen.dm]:
```suggestion
		to_chat(user, "<span class='notice'>Your helmet's torch is broken! You'll have to repair it with a lightbulb!</span>")
```

**Review by CydiaLamiales — COMMENTED** [review]
Changes

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 2 above)
Comment (CydiaLamiales) [reply code/modules/clothing/spacesuits/plasmamen.dm]: Sounds reasonable

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 3 above)
Comment (CydiaLamiales) [reply code/modules/clothing/spacesuits/plasmamen.dm]: Done

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/clothing/spacesuits/plasmamen.dm:225]: Then you should be able to lighteater it

**Comment by itsmeow** [comment]
Update CL

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/clothing/spacesuits/plasmamen.dm:225]: Oh. `helmet_on` means that the lamp itself is on. Nevermind. Badly named var lol

**Review by itsmeow — APPROVED** [review]

**Pull request merged**